pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -167,6 +167,29 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	return nil
 }
 
+// Test runs the devfile test command
+func (a Adapter) Test(testCmd string, show bool) (err error) {
+	pod, err := a.Client.GetPodUsingComponentName(a.ComponentName)
+	if err != nil {
+		return fmt.Errorf("no valid pod can be found to run 'odo test': %w", err)
```

Comment:
We shouldn't say `no valid pod can be found to run` as we couldn't determine it due to the error. This should be more like `error occurred while getting the pod...`

Reply:
updated